bears/uploadTool.py

```diff
@@ -0,0 +1,103 @@
+import glob
+import os
+import re
+import shutil
+import sys
+
+# inserting path here so the following imports work wherever you run the script
+# in coala-bears
+sys.path.insert(0, '..')
+
+from bears.Constants import CLASSIFIERS
+from bears.Constants import ENCODING
+from bears.Constants import IMPORTS
+from coalib.collecting.Importers import iimport_objects
+from string import Template
+
+
+def create_directory(name):
+    """
+    Checks if the folder exists, and creates it if it doesn't.
+
+    :param name: The directory name.
+    """
+    if not os.path.exists(name):
+        os.mkdir(name)
+
+
+def create_directory_hierarchy(folder_name, subfolder_name):
+    """
+    Checks if the folders do not exist and creates a folder hierarchy as in:
+    `folder_name/subfolder_name/subfolder_name`.
+
+    :param folder_name:    The folder in which the hierachy is created.
+    :param subfolder_name: The subfolder name.
+    """
+    if not os.path.exists(os.path.join(folder_name, subfolder_name)):
+        os.mkdir(os.path.join(folder_name, subfolder_name))
+    if not os.path.exists(os.path.join(folder_name, subfolder_name,
+                                       subfolder_name)):
+        os.mkdir(os.path.join(folder_name, subfolder_name, subfolder_name))
+
+
+def create_empty_file(file_name):
+    """
+    Creates an empty file.
+
+    :param file_name: Name of the file.
+    """
+    if not os.path.exists(file_name):
+        open(file_name, 'w').close()
+
+
+def main():
+    create_directory('upload')
+
+    bears_full_filenames = glob.glob('**/*Bear.py')
+
+    for bear_file_name in bears_full_filenames:
+        file_name = bear_file_name
+        bear_name = re.match(r'.+/(.*)\.py', bear_file_name).group(1)
+        bear_object = list(iimport_objects(file_name, attributes='kind',
+                                           local=True))
+        create_directory_hierarchy('upload', bear_name)
+        create_empty_file(os.path.join('upload', bear_name, bear_name,
+                                       '__init__.py'))
+        shutil.copyfile(file_name, os.path.join('upload', bear_name, bear_name,
+                                                bear_name + '.py'))
+
+        with open('setup.in', 'r') as fl:
+            template = Template(fl.read())
+
+        .. Start ignoring LineLengthBear
+
+        processed_template = template.safe_substitute(IMPORTS=IMPORTS,
+                                                      ENCODING=ENCODING, bear_name=bear_name, VERSION='1.0.30',
+                                                      AUTHORS=str(
+                                                          bear_object[0].AUTHORS),
```

Comment:
so it it always [0]? Why is it a list in the first place?

Might want to use **locals() here instead of explicitly handing over every template variable